### PR TITLE
WIP: Reworked based on Puppet 4 language and Hiera 5

### PR DESCRIPTION
**Pull request by genebean**
Tests need to be overhauled still but this should cover most, if not all, the code changes needed to update everything to using the facts hash, Puppet 4 data types, and in module data compatible with Hiera 5.

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -3,19 +3,22 @@
...
-  $settings = deep_merge($::autosign::params::settings, $::autosign::settings)
-
   file {$::autosign::configfile:
-    ensure  => $config_ensure,
+    ensure  => $_config_ensure,
     mode    => '0640',
-    content => template('autosign/autosign.conf.erb'),
-    owner   => $::autosign::user,
-    group   => $::autosign::group,
+    content => epp('autosign/autosign.conf.epp', {
```
Comment (danieldreier) [inline manifests/config.pp:14]: awesome! Love to see EPP here.

File: manifests/config.pp [hunk 2]
```diff
@@ -3,19 +3,22 @@
 # This class is called from autosign for service config.
 #
 class autosign::config {
-  $config_ensure = $::autosign::ensure ? {
+  $_config_ensure = $autosign::ensure ? {
```
Comment (danieldreier) [inline manifests/config.pp:6]: why omit the explicit top-scope namespacing here, going from `$::autosign::ensure` to plain `$autosign::ensure`? I'm not opposed, I'd just like to know if the convention here has changed.

File: manifests/install.pp [hunk 3]
```diff
@@ -2,24 +2,31 @@
...
 # This class is called from autosign for install.
 #
-class autosign::install {
-
-  # install the autosign gem
-  package { $::autosign::package_name:
-    ensure   => $::autosign::ensure,
-    provider => $::autosign::gem_provider,
+class autosign::install inherits autosign {
+  # install the autosign gem for both the agent and puppetserver
```
Comment (danieldreier) [inline manifests/install.pp:6]: this is great, fixes a long-standing frustration

**Review by danieldreier — APPROVED** [review]

File: manifests/config.pp (hunk 2 above)
Comment (genebean) [reply manifests/config.pp:6]: The docs no longer use top scope for anything other than classes inclusion / declarations.
https://puppet.com/docs/puppet/5.3/hiera_migrate_modules.html

The addition of the `_` prefix was a result of https://puppet.com/docs/puppet/5.0/lang_variables.html#naming

**Comment by danieldreier** [comment]
@genebean I really appreciate this work, but I'm closing it because https://github.com/danieldreier/puppet-autosign/pull/21 is already merged and covers the same ground.

**Comment by logicminds** [comment]
@genebean would you want to refactor this and make some of these changes again.  I am already looking at doing some of the same things and saw this PR.

**Comment by genebean** [comment]
Sure - whatcha want me to do?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request closed without merging**